### PR TITLE
ktm-49: Add eslint rules

### LINKED ISSUE #49: Add eslint rules

### PR DESCRIPTION
**Pull request by kuslosolodka**

### CODE REVIEW

File: backend/src/app.controller.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { Controller, Get } from '@nestjs/common';
 
+// eslint-disable-next-line import/extensions
```
Comment (HIMDanny) [inline backend/src/app.controller.ts:3]: I've noticed that this rule is repeated quite often. How about disabling it in the config?

File: backend/src/app.controller.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 import { AppService } from './app.service';
 
 @Controller()
+// eslint-disable-next-line no-restricted-syntax
```
Comment (HIMDanny) [inline backend/src/app.controller.ts:7]: This as well

File: client/src/App.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 function App() {
   return <div>Hello world</div>;
 }
 
+// eslint-disable-next-line import/no-default-export
```
Comment (HIMDanny) [inline client/src/App.tsx:6]: What is the problem with default export?

File: backend/src/app.controller.ts (hunk 1 above)
Comment (kuslosolodka) [reply backend/src/app.controller.ts:3]: This rule disabling for now. All disabled rules are now for a short time, later we figure out this

File: backend/src/app.controller.ts (hunk 2 above)
Comment (kuslosolodka) [reply backend/src/app.controller.ts:7]: https://github.com/HIMDanny/kanban-task-manager/pull/51#discussion_r1238274027

File: client/src/App.tsx (hunk 3 above)
Comment (kuslosolodka) [reply client/src/App.tsx:6]: https://www.lloydatkinson.net/posts/2022/default-exports-in-javascript-modules-are-terrible/. With naming exports you can more understand code structure

File: client/src/App.tsx [hunk 4]
```diff
@@ -1,5 +1,7 @@
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (HIMDanny) [inline client/src/App.tsx:1]: Typescript can automatically define return type. Don't see a necessity for the explicit return type

Comment (kuslosolodka) [reply client/src/App.tsx:1]: This rule enabling error if you don't define return type, not warning. Most rules already enabling with recommended syntax, but with warning. These configurations for enabling error, if you forgot setting return type. With errors lesser chances for passing eslint tests in commits, especially for "new-bies"

Comment (HIMDanny) [reply client/src/App.tsx:1]: I understand that it throws an error) I say that I don't see the necessity for this rule. I suggest turning this rule off.

Comment (kuslosolodka) [reply client/src/App.tsx:1]: (https://typescript-eslint.io/rules/explicit-function-return-type/)

> **However, explicit return types do make it visually more clear what type is returned by a function. They can also speed up TypeScript type checking performance in large codebases with many large functions.**

Comment (kuslosolodka) [reply client/src/App.tsx:1]: Most rules which we're using for our understading of code structure, by under hood there are already can be integrated

Comment (HIMDanny) [reply client/src/App.tsx:1]: Steel don't think that this is a necessary rule...

**Review by HIMDanny — APPROVED** [review]

File: client/src/App.tsx (hunk 4 above)
Comment (kuslosolodka) [reply client/src/App.tsx:1]: Doesn't everyone understanding typescript in our project :) this is improving learning and understanding of typescript

**Review by HIMDanny — APPROVED** [review]

**Review by HIMDanny — APPROVED** [review]

**Review by bymooon — APPROVED** [review]

**Pull request merged**